[Bug][Service] All tasks that need to send attachment will fail due to chennel idle

**Describe the bug**
When I execute a `sql` task and set to sent an email, the task will fail. But the email is sent successfully.
**To Reproduce**
Steps to reproduce the behavior, for example:
1. Create a `sql` task and set send email.
![image](https://user-images.githubusercontent.com/22415594/117470554-84e65500-af89-11eb-8768-70b1f6ef4e04.png)
2. execute
3. See error in work log and see task status in ui.
```java
[INFO] 2021-05-07 23:05:11.932 org.apache.dolphinscheduler.service.alert.AlertClientService:[100] - sync alert send, host : localhost, port : 50052, groupId : 2, title : Ds test
[ERROR] 2021-05-07 23:05:21.944 org.apache.dolphinscheduler.service.alert.AlertClientService:[110] - sync alert send error
org.apache.dolphinscheduler.remote.exceptions.RemotingTimeoutException: wait response on the channel Host{address='localhost:50052', ip='localhost', port=50052} timeout 10000
	at org.apache.dolphinscheduler.remote.NettyRemotingClient.sendSync(NettyRemotingClient.java:278)
	at org.apache.dolphinscheduler.service.alert.AlertClientService.sendAlert(AlertClientService.java:105)
	at org.apache.dolphinscheduler.service.alert.AlertClientService.sendAlert(AlertClientService.java:87)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.sendAttachment(SqlTask.java:458)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.resultProcess(SqlTask.java:332)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:264)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.handle(SqlTask.java:153)
	at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:169)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[ERROR] 2021-05-07 23:05:21.945  - [taskAppId=TASK-8-211-220]:[276] - execute sql error: null
[ERROR] 2021-05-07 23:05:21.945  - [taskAppId=TASK-8-211-220]:[159] - sql task error: java.lang.NullPointerException
[ERROR] 2021-05-07 23:05:21.945 org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread:[182] - task scheduler failure
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.sendAttachment(SqlTask.java:459)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.resultProcess(SqlTask.java:332)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:264)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.handle(SqlTask.java:153)
	at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:169)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```


**Which version of Dolphin Scheduler:**
 -[dev]

**Additional context**
This is mainly caused by the following code, you can find in `NettyRemotingClient`. The `responseFuture.waitResponse()` return null, and this method will throw exception.
```java
final ResponseFuture responseFuture = new ResponseFuture(opaque, timeoutMillis, null, null);
channel.writeAndFlush(command).addListener(future -> {
    if (future.isSuccess()) {
        responseFuture.setSendOk(true);
        return;
    } else {
        responseFuture.setSendOk(false);
    }
    responseFuture.setCause(future.cause());
    responseFuture.putResponse(null);
    logger.error("send command {} to host {} failed", command, host);
});
/*
 * sync wait for result
 */
Command result = responseFuture.waitResponse();
if (result == null) {
    if (responseFuture.isSendOK()) {
        throw new RemotingTimeoutException(host.toString(), timeoutMillis, responseFuture.getCause());
    } else {
        throw new RemotingException(host.toString(), responseFuture.getCause());
    }
}
```
And this actually caused by the `AlertServer` didn't return the response to the worker, the possible reason is that the channel closed due to the long time no data transmission. When I increase the `readerIdleTime` from 6s to 10s, this issue can be solved.

**Requirement or improvement**
1. Add `readerIdleTime` configuration in clientConfig, some client like Alert needs larger idleTime.
2. When the email send error, the task shouldn't be affected

--- a/dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyClientHandler.java
+++ b/dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyClientHandler.java
@@ -184,9 +184,11 @@ public class NettyClientHandler extends ChannelInboundHandlerAdapter {
             Command heartBeat = new Command();
             heartBeat.setType(CommandType.HEART_BEAT);
             heartBeat.setBody(heartBeatData);
-            ctx.writeAndFlush(heartBeat)
+            ctx.channel().writeAndFlush(heartBeat)
                     .addListener(ChannelFutureListener.CLOSE_ON_FAILURE);
-
+            if (logger.isDebugEnabled()) {
+                logger.debug("Client send heart beat to: {}", ChannelUtils.getRemoteAddress(ctx.channel()));
+            }
         } else {
             super.userEventTriggered(ctx, evt);
         }

--- a/dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java
+++ b/dolphinscheduler-remote/src/main/java/org/apache/dolphinscheduler/remote/handler/NettyServerHandler.java
@@ -116,6 +116,12 @@ public class NettyServerHandler extends ChannelInboundHandlerAdapter {
      */
     private void processReceived(final Channel channel, final Command msg) {
         final CommandType commandType = msg.getType();
+        if (CommandType.HEART_BEAT.equals(commandType)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("server receive heart beat from: host: {}", ChannelUtils.getRemoteAddress(channel));
+            }
+            return;
+        }
         final Pair<NettyRequestProcessor, ExecutorService> pair = processors.get(commandType);
         if (pair != null) {
             Runnable r = () -> {

